File: 2025_08/leetcode2025-08-13.py
from math import log

class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        return False if n<=0 else pow(3, round(log(abs(n), 3)))==n

if __name__ == "__main__":
    s = Solution()
    n = 27
    print(s.isPowerOfThree(n))

File: leetcode2025/2025_08/leetcode2025-08-13.cpp
#include <iostream>
#include <cmath>
using namespace std;

class Solution {
public:
    bool isPowerOfThree(int n) {
        int pows = 0, origin = n;
        while (n>=3){
            n /= 3;
            pows++;
        }
        return pow(3, pows)==origin;
    }
};

int main(){
    Solution s;
    int n = 27;
    cout << s.isPowerOfThree(n) << endl;
}
